### ISSUE TITLE
Cannot access attributes of nested object

### ISSUE THREAD

**Issue by wjdhamilton** [ncri/nested_form_fields#55]
When trying to gain access to the attributes of an associated object whilst updating its parent, it is possible to access the associated object, but not its attributes. For instance, the following works:

```` 
<%= f.nested_fields_for :charges do |cf| %>
          <%= cf.object.account %> 
      <% end %>
````

Whereas this fails with `undefined method 'id' for nil:NilClass`:

```` 
<%= f.nested_fields_for :charges do |cf| %>
          <%= cf.object.account.id %> 
 <% end %>
````

Laws of Demeter aside, this feels profoundly odd.

**Comment by ncri** [comment]
It does indeed :) what does cf.object give you exactly? A Charge instance would be expected.

**Comment by wjdhamilton** [comment]
Yes, it gives a Charge instance. And the correct one too! It's as though Rails doesn't go to the database to find associated objects if they haven't been loaded.

**Comment by ncri** [comment]
I have never seen this behavior. most likely it has nothing to do with nested_form_fields, as nested_form_fields simply sits on top of the basic rails form builder. Can you try to remove nested_form_fields and see if the issue is still there?

**Comment by wjdhamilton** [comment]
I can confirm that the issue is not there if I use fields_for.

**Comment by ncri** [comment]
oh, that is really odd... i need to check if nested_form_fields does anything different here than fields_for... strange, i didn't expect that

**Comment by ncri** [comment]
they are blocked now :) they wont get through to the app, just see a
"forbidden" page

**Comment by ncri** [comment]
oops, sorry for the unrelated comment, i accidentally replied to the wrong
email ;-)

**Comment by wjdhamilton** [comment]
LOL I was struggling to put that one into context!

**Comment by ncri** [comment]
happened to me the first time ;-)

**Comment by ncri** [comment]
Well, at least for now you can use a workaround: 

    - account_id = cf.object.account_id

or, if you need the whole account object:

    - account = Account.find(cf.object.account_id)

**Comment by wjdhamilton** [comment]
I swear I tried the first option without success but it's working now! Thanks for your help and attention :+1:

**Comment by bassman88** [comment]
I'm currently facing the same problem. It only works on 1 parent model association, the other two, no luck: 

`<%= form.nested_fields_for :delivery_items, wrapper_tag: :tr do |ff| %>
  <td class="col-md-1">
    <div class="input-product_code-div">
        <%= ff.object.product.name  %> <br> #works!
        <%= ff.object.sales_order_item.address  %> <br> #does not work
        <%= ff.object.delivery.id  %> #does not work
        <%end%>
    </div>
  </td>
<% end %>`

**Comment by ncri** [comment]
@bassman88: does the workaround work for you?
